Build of benchmark test fails with gcc 14.2.0

I try to run the benchmarks test on macOS using gcc 14.2.0, I get:

```
$ tests bench -r -c M1
INFO  > Benchmark          Compile     (native, no_opt) make CROSS_PREFIX= bench CYCLES=M1 AUTO=1 OPT=0
test/hal/hal.c: In function ‘init_rdtsc’:
test/hal/hal.c:250:10: error: ISO C forbids conversion of object pointer to function pointer type [-Werror=pedantic]
  250 |   name = (name##proc *)(dlsym(kperf, #name)); \
      |          ^
test/hal/hal.c:159:3: note: in expansion of macro ‘F’
  159 |   F(int, kpc_get_counting, void)               \
      |   ^
test/hal/hal.c:256:3: note: in expansion of macro ‘KPERF_LIST’
  256 |   KPERF_LIST
      |   ^~~~~~~~~~
```

Same thing using Apple's default /usr/bin/gcc (which is actually clang 16.0.0) works fine.

--- a/test/hal/hal.c
+++ b/test/hal/hal.c
@@ -246,6 +246,13 @@ static void init_rdtsc(void)
     printf("kperf = %p\n", kperf);
     return;
   }
+
+/* Temporarily disable -Wpedantic here to allow conversion
+ * from (void *) to function-pointer.
+ */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpedantic"
+
 #define F(ret, name, ...)                     \
   name = (name##proc *)(dlsym(kperf, #name)); \
   if (!name)                                  \
@@ -255,6 +262,8 @@ static void init_rdtsc(void)
   }
   KPERF_LIST
 #undef F
+
+#pragma GCC diagnostic pop
 
   g_config[0] = CPMU_CORE_CYCLE | CFGWORD_EL0A64EN_MASK;
 }

